proto/babylon/zoneconcierge/v1/zoneconcierge.proto

```diff
@@ -0,0 +1,148 @@
+syntax = "proto3";
+package babylon.zoneconcierge.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+import "tendermint/crypto/proof.proto";
+import "babylon/btccheckpoint/v1/btccheckpoint.proto";
+import "babylon/checkpointing/v1/bls_key.proto";
+import "babylon/checkpointing/v1/checkpoint.proto";
+import "babylon/epoching/v1/epoching.proto";
+import "babylon/btclightclient/v1/btclightclient.proto";
+
+option go_package = "github.com/babylonlabs-io/babylon/x/zoneconcierge/types";
+
+// IndexedHeader is the metadata of a CZ header
+message IndexedHeader {
+  // consumer_id is the unique ID of the consumer
+  string consumer_id = 1;
+  // hash is the hash of this header
+  bytes hash = 2;
+  // height is the height of this header on CZ ledger
+  // (hash, height) jointly provides the position of the header on CZ ledger
+  uint64 height = 3;
+  // time is the timestamp of this header on CZ ledger
+  // it is needed for CZ to unbond all mature validators/delegations
+  // before this timestamp when this header is BTC-finalised
+  google.protobuf.Timestamp time = 4 [ (gogoproto.stdtime) = true ];
+  // babylon_header_hash is the hash of the babylon block that includes this CZ
+  // header
+  bytes babylon_header_hash = 5;
+  // babylon_header_height is the height of the babylon block that includes this CZ
+  // header
+  uint64 babylon_header_height = 6;
+  // epoch is the epoch number of this header on Babylon ledger
+  uint64 babylon_epoch = 7;
+  // babylon_tx_hash is the hash of the tx that includes this header
+  // (babylon_block_height, babylon_tx_hash) jointly provides the position of
```

Comment:
what happens if in the same tx two headers are indexed?

Reply:
Good question. It's fine actually as we have consumer ID as the KV store prefix ([ref](https://github.com/babylonlabs-io/babylon/blob/873f1232365573a97032037af4ac99b5e3fcada8/x/zoneconcierge/keeper/canonical_chain_indexer.go#L59-L62))

Reply:
I mean two headers for the same consumer/height/tx (as tx can contain multiple msgs) being indexed 

Seems like we are only overwriting it by the last one that passes

Reply:
Hmm I see what you mean, but in this case wouldn't IBC modules reject the duplicated submission before the tx goes to the PostHandler?

Reply:
not sure, it is worth to test it out

Reply:
Duplicated header will get rejected, see https://github.com/cosmos/ibc-go/blob/91169d134d659b87acaf4903652cdcb1a55f5657/modules/light-clients/07-tendermint/update.go#L85-L91